content/post/practical-introR.Rmd

```diff
@@ -0,0 +1,697 @@
+
+matrix_of_int <-  matrix(data = vector_integer, nrow = 4, ncol = 5)
+matrix_of_int
+
+
+dim(matrix_of_int)
+
+```
+
+
+
+### Arrays
+
+Array is a special type of matrix, where there is more than two dimensions (n dimensions).
+
+An array of two dimensions is a matrix\
+
+To create an array we need three arguments: *data* and *dim*.
+
+In turn, the *dim* argument of an array is composed of three arguments being: 1) number of rows, 2) number of columns and 3) number of dimensions.
+
+
+
+```{r}
+
+vector_example <-1:18
+array_example <- array(data = vector_example, dim = c(2,3,3))
+
+dim(array_example)
+array_example
+
+```
+
+
+
+### Dataframes
+
+A *dataframe* is a heterogeneous and bidimensional structure, similar but not exactly equal to a matrix. Unlike a matrix, various types of vectors can be part of a single dataframe.
+
+The arguments for the *data.frame* command are simply the columns in the dataframe. Each column shoud have the same number of rows to bea able to fit into a daraframe.
+
+Data frames do not allow vectors with different lengths. When the length of the vector is smaller than the length of the data frame, the data frame coerces the vector to its length.
+
+```{r}
+
+
+data_example <- data.frame( vector_character,vector_double, vector_logic, vector_integer)
+
+```
+
+To access the general structure of a dataframe we use the command *str* 
+
+```{r}
+
+str(data_example)
+
+```
+
+
+To access the different components of the dataframe we use this syntax *[,]* where the first dimension corresponds to rows and the second dimension to columns.
+
+```{r}
+
+data_example[1,2]
+
+```
+
+
+
+### List
+
+A *list* is the most complex structure in base R. A list cab be composed of any type of other structures of any dimension.
+
+```{r}
+
+list_example <- list(vector_character,
+                     matrix_of_int,
+                     data_example)
+
+```
+
+
+To access the different components of a list, we use the syntax *[]* where the argument is simply the order within the list.
+
+```{r}
+
+list_example[1]
+
+```
+
+
+
+
+# Functions
+
+A function is one of the structures that makes *R* a very powerful platform for programming. 
+
+There are varios types of functions
+
+- *Primitive or base functions*: these are the default functions in _R_ under the _base_ package. For instance, for basic arithmentic operations such as addition _sum_, substraction _*_, multiplication _*_, but also for more complex operations such as extraction of median values _median_, or even more complex like _summary_ of a variable.
+
+- *Functions from packages*: these are functions created within a package. For example the function *glm* in the *stats* package.
+
+
+- *User-built functions*: these are the functions that any user creates for a customised routine. In turn, these functions could become part of a package.
+
+
+The basic components of a function are:
+
+
+- __*name*:__ this is the name that we give to our function (for example: _myfun_)
```

Comment:
This is showing the stars and bold. 

`__*name*__`: __*name*__
`*__name__*`: *__name__*
`***name***`: ***name***
`___name___`: ___name___